tests/ert/unit_tests/config/test_forward_model.py

```diff
@@ -92,190 +92,112 @@ def test_load_forward_model_upgraded():
-        f.write("This is a script")
-        name = "script.sh"
-        with open(name, "w", encoding="utf-8") as f:
-            f.write("This is a script")
-        mode = os.stat(name).st_mode
-        mode |= stat.S_IXUSR | stat.S_IXGRP
-        os.chmod(name, stat.S_IMODE(mode))
     with (
         pytest.raises(ConfigValidationError, match="EXECUTABLE must be set"),
         pytest.warns(ConfigWarning, match='"PORTABLE_EXE" key is deprecated'),
     ):
-        _ = _forward_model_step_from_config_file("CONFIG")
+        _ = _forward_model_step_from_config_contents(
+            "PORTABLE_EXE fm_dispatch.py", "CONFIG"
+        )
 
 
-@pytest.mark.usefixtures("use_tmpdir")
 def test_load_forward_model_missing_raises():
-    with open("CONFIG", "w", encoding="utf-8") as f:
-        f.write("EXECUTABLE missing_script.sh\n")
     with pytest.raises(ConfigValidationError, match="Could not find executable"):
-        _ = _forward_model_step_from_config_file("CONFIG")
+        _ = _forward_model_step_from_config_contents(
+            "EXECUTABLE missing_script.sh", "CONFIG"
+        )
 
 
 @pytest.mark.filterwarnings("ignore:.*Unknown keyword 'EXECU'.*:UserWarning")
-@pytest.mark.usefixtures("use_tmpdir")
 def test_load_forward_model_execu_missing_raises():
-    with open("CONFIG", "w", encoding="utf-8") as f:
-        f.write("EXECU missing_script.sh\n")
     with pytest.raises(ConfigValidationError, match="EXECUTABLE must be set"):
-        _ = _forward_model_step_from_config_file("CONFIG")
+        _ = _forward_model_step_from_config_contents(
+            "EXECU missing_script.sh\n", "CONFIG"
+        )
 
 
-@pytest.mark.usefixtures("use_tmpdir")
 def test_load_forward_model_is_directory_raises():
-    with open("CONFIG", "w", encoding="utf-8") as f:
-        f.write("EXECUTABLE /tmp\n")
     with pytest.raises(ConfigValidationError, match="directory"):
-        _ = _forward_model_step_from_config_file("CONFIG")
+        _ = _forward_model_step_from_config_contents("EXECUTABLE /tmp", "CONFIG")
 
 
-@pytest.mark.usefixtures("use_tmpdir")
 def test_load_forward_model_foreign_raises():
-    with open("CONFIG", "w", encoding="utf-8") as f:
-        f.write("EXECUTABLE /etc/passwd\n")
     with pytest.raises(ConfigValidationError, match="File not executable"):
-        _ = _forward_model_step_from_config_file("CONFIG")
+        _ = _forward_model_step_from_config_contents("EXECUTABLE /etc/passwd", "CONFIG")
 
 
-def test_forward_model_optionals(
-    tmp_path,
-):
-    executable = tmp_path / "exec"
-    executable.write_text("")
-    st = os.stat(executable)
-    os.chmod(executable, st.st_mode | stat.S_IEXEC)
-    config_file = tmp_path / "config_file"
-    config_file.write_text("EXECUTABLE exec\n")
-    forward_model = _forward_model_step_from_config_file(str(config_file))
+def test_forward_model_optionals():
+    forward_model = _forward_model_step_from_config_contents(
+        "EXECUTABLE fm_dispatch.py", "config_file"
+    )
```

Comment:
This test tests the same as the one below. Perhaps this is superfluous.

Reply:
Removed